rules/dotnet-upgrade-for-complex-projects/rule.md

```diff
@@ -1,169 +1,203 @@
+
+This will create a new .NET 8 project. For functionalities that have not yet been migrated, YARP will redirect them to the .NET Framework web application.
+
+### Configure YARP
+
+```csharp
+var webRoutes = new List<RouteConfig>
+{
+    // Route for token
+    new()
+    {
+        RouteId = "tokenServePath",
+        ClusterId = tokenClusterId,
+        Match = new RouteMatch
+        {
+            Path = "/token/{**catch-all}",
+        },
+    },
+
+    // Route for WebUI App
+    new RouteConfig
+    {
+        RouteId = "webUIServePath",
+        ClusterId = webUiClusterId,
+        Match = new RouteMatch
+        {
+            Path = "/api/v2/{**catch-all}",
+        },
+    },
+
+    // Route for WebApp App
+    new RouteConfig
+    {
+        RouteId = "webAppServePath",
+        ClusterId = webAppClusterId,
+        Match = new RouteMatch
+        {
+            Path = "/api/{**catch-all}",
+        },
+    },
+
+    // Route for Angular
+    new RouteConfig
+    {
+        RouteId = "angularUIServePath",
+        ClusterId = angularClusterId,
+        Match = new RouteMatch
+        {
+            Path = "{**catch-all}",
+        },
+    }
+};
+
+```
+**Figure: Example code for setting up different paths within YARP's configuration.**
+
+
+### Upgrading components using Upgrade Assistant
+
+Once you have created the side-by-side project, select the project that needs migration and `right click` | `Upgrade` on it.
+
+![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/3699937/3303daaf-0dea-4b34-9f59-53fd55acf2ef)
+
+**Figure: Context menu on the project to be migrated.**
+
+Upgrade Assistant will show you a Summary view and detect that the project is linked to your Yarp proxy. 
+You can also see the migration progress of your endpoints from .NET Framework to .NET as a pie chart.
+
+![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/3699937/8564c9a7-b3a7-4b40-b002-be9c6fabcb16)
+
+**Figure: Upgrade Assistants Summary page.**
+
+
+From here you can explore your endpoints through the `Endpoint explorer`, which will also indicate what endpoints have already been migrated and which ones are still outstanding.
+The chain icon ![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/3699937/d89d7150-e0b3-4947-abeb-0e1f865ab6f8) indicates that this endpoint has been migrated and is linked between
+the controller in the old project and the controller in the Yarp proxy project.
+
+![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/3699937/15377711-45a9-41dd-88b5-c555b64e6a87)
+
+**Figure: Endpoint Explorer showing the endoints between the old .NET Framework project and the new .NET Core project.**
+
+Use the `Upgrade` functionality for example the `Upgrade Controller` to let Upgrade Assistant help you with the migration by analyzing and applying automatic code transformations to speed up the process. 

```

Comment:
I think this sentence could be improved. I suggest skipping the example reference and rewording this sentence to something like:

> "Use the `Upgrade` functionality to apply automatic code transformations and speed up the migration process."

Reply:
applied chages.